autoload/lsp.vim

```diff
@@ -507,6 +512,7 @@ function! s:ensure_changed(buf, server_name, cb) abort
     let l:buffer_info['changed_tick'] = l:changed_tick
     let l:buffer_info['version'] = l:buffer_info['version'] + 1
 
+    call lsp#ui#vim#folding#send_request(a:server_name, a:buf, 0)
```

Comment:
Sending before every `textDocument/didChange` doesn't work that well. If you're halfway with typing e.g.
```javascript
function f() {
```
then the language server doesn't give correct folds, leading to the folds in Vim jumping all over the place.
How about we only send requests on `CursorHold` autocmd and when no errors are present in the buffer?